manager/scheduler/scheduler_test.go

```diff
@@ -2098,6 +2167,42 @@ func TestSchedulerPluginConstraint(t *testing.T) {
 	assignment2 := watchAssignment(t, watch)
 	assert.Equal(t, assignment2.ID, "task3_ID")
 	assert.Equal(t, assignment2.NodeID, "node3_ID")
+
+	// Create t4; it should stay in the pending state because there is
+	// no node that with log plugin `plugin1`
+	err = s.Update(func(tx store.Tx) error {
+		assert.NoError(t, store.CreateTask(tx, t4))
+		return nil
+	})
+	assert.NoError(t, err)
+
+	time.Sleep(100 * time.Millisecond)
+	s.View(func(tx store.ReadTx) {
+		task := store.GetTask(tx, "task4_ID")
+		if task.Status.State >= api.TaskStateAssigned {
+			t.Fatal("task 'task4_ID' should not have been assigned")
+		}
+	})
```

Comment:
Since this PR was opened, the other scheduler tests were updated to use `watchAssignmentFailure`, so I'd appreciate it if you could update this one to use it as well. Then this will be ready to merge.